### ISSUE TITLE
.NET 7 Preview 7 - Polymorphic Deserialization ignores PropertyNameCaseInsensitive setting

### ISSUE THREAD

**Issue by SefaOray** [dotnet/runtime#75269]
### Description

I have the following class structure

```
[JsonPolymorphic(TypeDiscriminatorPropertyName = nameof(ActivityType))]
[JsonDerivedType(typeof(LearnSkillActivity), (int) CareerActivities.LearnSkill)]
public abstract class BaseProfessionalActivity
{
     public abstract CareerActivities ActivityType { get; }
}

public class LearnSkillActivity : BaseProfessionalActivity
{
    public string SkillId { get; set; }

    public override CareerActivities ActivityType => CareerActivities.LearnSkill;
}
```

### Reproduction Steps

Deserializing following json doesn't work

```
{
  "activityType": 0,
  "skillId": "63189799bc0a84a02ffc4bc9"
}
```

while the following works

```
{
  "ActivityType": 0,
  "skillId": "63189799bc0a84a02ffc4bc9"
}
```

Mind that only difference is the capical A in `ActivityType`

### Expected behavior
Deserializes correctly without error

### Actual behavior

Throws error

`System.NotSupportedException: Deserialization of types without a parameterless constructor, a singular parameterized constructor, or a parameterized constructor annotated with 'JsonConstructorAttribute' is not supported`

### Regression?
No response

### Known Workarounds
No response

### Configuration
> dotnet --version
7.0.100-preview.7.22377.5

### Other information
No response

**Comment by eiriktsarpalis** [comment]
I tend to agree that this is an omission that should be fixed. However, it doesn't meet the bar for a .NET 7 fix.

**Comment by eiriktsarpalis** [comment]
Moving to Future as we won't be able to work on this for 8.0

**Comment by iPilot** [comment]
Serializer should also consider case options for discriminator property on serialization too. For now, discriminator property name is written as specified in attribute or contract model configuration.

**Comment by Mortana89** [comment]
This is blocking our upgrade from .net 5 using Dahomey JSON polymorphism to .net 8 STJ.
Our old frontend is using lowercase type discriminators but the new front-end/any other API integration works with camelcasing. How can we work around this? Was trying to implement a custom resolver and duplicating the derived types with lowercased discriminators, but the SDK does not allow registering the same type twice

**Comment by eiriktsarpalis** [comment]
Unlikely at his point, unfortunately.
